services/events/project/api/events.py

```diff
@@ -145,10 +144,16 @@ def get_all_events():
     """Get all events"""
 
     current_time = datetime.utcnow()
-    events = Event.query.filter(Event.time > current_time).all()
+    recent_past = current_time - timedelta(hours=6)
+
+    upcoming_events = Event.query.filter(Event.time > current_time).all()
+    recent_events = Event.query.filter(Event.time <= current_time).filter(Event.time >= recent_past).all()
```

Comment:
The following should be valid here, and slightly more efficient when converted to SQL under the hood.
```
   and_(
        Event.time <= current_time,
        Event.time >= recent_past
    )
```